mabwiser/linear.py

```diff
@@ -208,40 +205,38 @@ def _fit_arm(self, arm: Arm, decisions: np.ndarray, rewards: np.ndarray, context
 
     def _predict_contexts(self, contexts: np.ndarray, is_predict: bool,
                           seeds: Optional[np.ndarray] = None, start_index: Optional[int] = None) -> List:
+        pass
+
+    def _vectorized_predict_context(self, contexts: np.ndarray, is_predict: bool) -> List:
 
-        # Get local copy of model, arm_to_expectation and arms to minimize
-        # communication overhead between arms (processes) using shared objects
-        arm_to_model = deepcopy(self.arm_to_model)
-        arm_to_expectation = deepcopy(self.arm_to_expectation)
+        # Converting the arms list to numpy array
         arms = deepcopy(self.arms)
+        arms = np.array(arms)
 
-        # Create an empty list of predictions
-        predictions = [None] * len(contexts)
-        for index, row in enumerate(contexts):
-            # Each row needs a separately seeded rng for reproducibility in parallel
-            rng = create_rng(seed=seeds[index])
+        # Initializing array with expectations for each arm
+        num_contexts = contexts.shape[0]
+        arm_expectations = np.empty((num_contexts, len(arms)), dtype=float)
 
-            # With epsilon probability set arm expectation to random value
-            if rng.rand() < self.epsilon:
-                for arm in arms:
-                    arm_to_expectation[arm] = rng.rand()
+        # With epsilon probability, assign random flag to context
```

Comment:
does this mean behavior of existing test/code will change? Since we are changing from 1by1 random setc?

Reply:
No, the behaviour would be the same from the algorithmic point of view. The reason we were using a random state for each context was to factor in the seed values for multi-processing module for results reproducibility. Since, we have eliminated the need for multi-processing using NumPy vectorization, we can just use one global seed value for reproducibility.

Reply:
Gotch, by _behavior_ I meant the output and specific cases. I see that tests are tweaked a little with the new behavior from one global seed. Cool!